src/Module/BaseApi.php

```diff
@@ -66,6 +94,29 @@ public static function post(array $parameters = [])
 		}
 	}
 
+	public static function put(array $parameters = [])
+	{
+		if (!api_user()) {
+			throw new HTTPException\UnauthorizedException(DI::l10n()->t('Permission denied.'));
+		}
+
+		$a = DI::app();
+
+		if (!empty($a->user['uid']) && $a->user['uid'] != api_user()) {
+			throw new HTTPException\ForbiddenException(DI::l10n()->t('Permission denied.'));
+		}
+	}
+
+	public static function unsupported(string $method = 'all')
+	{
+		$path = DI::args()->getQueryString();
+		Logger::info('Unimplemented API call', ['path' => $path, 'method' => $method]);
+		$error = DI::l10n()->t('API endpoint %s "%s" is not implemented', $method, $path);
```

Comment:
Please put the HTTP method in uppercase which removes the need for quotes around the path. And remember to run `bin/run_xgettext.sh` after you're done with the change. 😉
```suggestion
		$error = DI::l10n()->t('API endpoint %s %s is not implemented', strtoupper($method), $path);
```